bin/oneoff/aichat_requests_to_csv

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env ruby
+
+require 'csv'
+require_relative '../../dashboard/config/environment'
+
+def get_substring_in_between(text, begin_string, end_string)
+  text.split(begin_string).last.split(end_string).first
+end
+
+def valid_json?(string)
+  !!JSON.parse(string)
+rescue JSON::ParserError
+  false
+end
+
+def get_blocked_by_details(blocked_by, details)
+  if details.class != Hash
+    return details
+  end
+  case blocked_by
+  when "openai"
+    return {
+      evaluation: details[:evaluation]
+    }.to_json
+  when "comprehend"
+    return {
+      flagged_text: details[:flagged_segment],
+      toxicity: details[:toxicity],
+      max_category: details[:max_category][:name],
+      score: details[:max_category][:score]
+    }.to_json
+  when "blocklist"
+    return {flagged_text: details[:blocked_word]}.to_json
+  when "webpurify"
+    return {flagged_text: details[:content]}.to_json
+  else
+    return {}.to_json
+  end
+end
+
+def main
+  options = {}
+  OptionParser.new do |opts|
+    opts.on('--section_id SECTION_ID') do |section_id|
+      options[:section_id] = Integer(section_id)
+    end
+  end.parse!
+
+  CSV.open("aichat_requests.csv", "w") do |csv|
+    headers = %w[
+      aichat_request_id user_id username name path created_at updated_at model_id role chat_message status blocked_by block_by_details
+    ]
+    csv << headers
+
+    student_users_in_section = Section.find(options[:section_id]).followers.map(&:student_user)
+    teacher_users_in_section = Section.find(options[:section_id]).active_section_instructors.map(&:instructor)
+    all_users_in_section = student_users_in_section + teacher_users_in_section
+    # start_date is when we began using OpenAI for safety layer.
+    start_date = DateTime.new(2024, 9, 26, 0, 0, 0)
+    end_date = DateTime.now
+    time_range = start_date..end_date
+    rows = AichatRequest.where(
+      user: all_users_in_section,
+      created_at: time_range,
+    )
+
+    rows.each do |row|
+      # In the case a data row does not include level_id (indicates an invalid test request).
+      next unless row.level_id.is_a? Integer
```

Comment:
@sanchitmalhotra126 - added this after your last look at PR. Excludes invalid test requests that don't include `level_id`.